code/game/machinery/jukebox.dm

```diff
@@ -275,3 +275,38 @@
 	icon_state = state_base
 	if(playing)
 		AddOverlays("[state_base]-running")
+
+/obj/machinery/media/jukebox/gramophone/MouseDrop(atom/over, src_location, over_location, src_control, over_control, params)
+	. = ..()
+	if(use_check(usr) || !Adjacent(usr))
+		return
+	if(!ishuman(usr) && (!isrobot(usr) || isDrone(usr))) //Humans and borgs can collapse, but not drones
+		return
+	new /obj/item/gramophone(get_turf(src))
```

Comment:
You aren't checking if the drop is done ontop of the mob, any drop on anything would be valid for this proc; also the drop should probably put the newly made item in hand, rather than on the floor, once dropped onto the mob